src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -271,6 +271,11 @@ public override void ToExpression(StringBuilder sb, FormulaValueSerializerSettin
             }
 
             sb.Append("}");
+        }
+
+        internal virtual void ShallowCopyFieldInPlace(string fieldName)
+        {
+            throw new NotImplementedException();

```

Comment:
>throw new NotImplementedException(); [](http://example.com/codeflow?start=12&length=36)

should this method just be abstract?

Reply:
I'm assuming we can't get the exception at runtime; so hw do we know *not* to call it in cases like UserInfoRecord?

Reply:
I created a separate interface for this, IMutationCopyField.  The binder should prevent mutation of a value which does not support this, but if not, the interface will cast and throw if not found in EvalVisitor.